### PR TITLE
Properly define and test course abilities

### PR DESCRIPTION
**Pull request by allenwq**
Depends on #679

### CODE REVIEW

**Comment by lowjoel** [comment]
You have a typo in the commit message...

**Comment by lowjoel** [comment]
Oh and a rebase please.

File: app/models/components/course/course_ability_component.rb [hunk 1]
```diff
@@ -17,8 +17,7 @@ def define_permissions
 
   def allow_showing_open_courses
     # TODO: Replace with just the symbols when Rails 5 is released.
-    can [:read, :register], Course, status: [Course.statuses[:published], Course.statuses[:opened],
-                                             'published', 'opened']
+    can [:read, :register], Course, status: [Course.statuses[:published], Course.statuses[:opened]]
```
Comment (lowjoel) [inline app/models/components/course/course_ability_component.rb]: I thought the cancancan thing allows just using symbols?

**Comment by lowjoel** [comment]
Don't worry about coverage, I'll fix that line.

**Comment by allenwq** [comment]
@lowjoel Fixed, sorry for the typo in commit message, there is no grammar check in vi :(

File: app/models/components/course/course_ability_component.rb (hunk 1 above)
Comment (allenwq) [inline app/models/components/course/course_ability_component.rb]: I tested before, not allowing.

Comment (allenwq) [inline app/models/components/course/course_ability_component.rb]: https://github.com/CanCanCommunity/cancancan/commit/5b361a402e4bbe9c03bc0a3ff285aea2e572a917#diff-f6e4fbc507ba12eef3ca788ea825ed02R58
Cancancan also uses the same form in their tests

Comment (lowjoel) [inline app/models/components/course/course_ability_component.rb]: OK, hope Rails 5 solves it.

**Pull request merged**